torch_xla/csrc/xla_sharding_util.cpp

```diff
@@ -382,4 +383,55 @@ std::vector<at::Tensor> ShardingUtil::ShardTensor(
   return shards;
 }
 
+void ShardingUtil::PrepareOutputShardingPropagation(
+    std::vector<XLATensorPtr>* tensors, absl::Span<const size_t> indices,
+    ComputationPtr computation,
+    std::vector<torch::lazy::BackendDataPtr>* data_placeholders,
+    std::vector<XLATensor::ShardingSpecPtr>* sharding_specs) {
+  auto computation_proto = computation->computation().proto();
+  auto tuple_shardings =
+      computation_proto.spmd_output_sharding().tuple_shardings();
+  if (tuple_shardings.empty()) {
+    return;
+  }
+  XLA_CHECK(indices.size() == tuple_shardings.size())
+      << "Expected " << indices.size() << " sharding annotations, but got "
+      << tuple_shardings.size();
+
+  for (int i = 0; i < indices.size(); ++i) {
+    auto xtensor = (*tensors)[indices[i]];
+    std::cout << "*** " << xtensor->CurrentIrValue()->ToString() << " sharded? "
```

Comment:
Log?

Reply:
The PR's in draft -- I am testing a few extra bits with probing. Will ping you when it's ready :) thanks.

Reply:
Ok removed the `std::cout`, I will mark this PR as ready-for-review tmr morning.